beacon-chain/state/stategen/getter.go

```diff
@@ -152,3 +148,144 @@ func (s *State) recoverStateSummary(ctx context.Context, blockRoot [32]byte) (*p
 	}
 	return nil, errUnknownStateSummary
 }
+
+// This loads a beacon state from either the cache or DB then replay blocks up the requested block root.
+func (s *State) loadStateByRoot(ctx context.Context, blockRoot [32]byte) (*state.BeaconState, error) {
+	ctx, span := trace.StartSpan(ctx, "stateGen.loadStateByRoot")
+	defer span.End()
+
+	// First, it checks if the state exists in hot state cache.
+	cachedState := s.hotStateCache.Get(blockRoot)
+	if cachedState != nil {
+		return cachedState, nil
+	}
+
+	// Second, it checks if the state exits in epoch boundary state cache.
+	cachedInfo, ok, err := s.epochBoundaryStateCache.getByRoot(blockRoot)
+	if err != nil {
+		return nil, err
+	}
+	if ok {
+		return cachedInfo.state, nil
+	}
+
+	summary, err := s.stateSummary(ctx, blockRoot)
```

Comment:
Can this be nil?

Reply:
It's gated here https://github.com/prysmaticlabs/prysm/blob/master/beacon-chain/state/stategen/getter.go#L153